src/gg.regression.unity.bots/Editor/Scripts/CodeGenerators/RGCodeGenerator.cs

```diff
@@ -107,818 +114,872 @@ public static void GenerateRGScripts()
-                                        continue;
-                                    }
-
-                                    break;
+                                    // is string
+                                    entityTypeName = arg0.Token.ValueText;
                                 }
-                                default:
-                                    // no methods
-                                    continue;
                             }
-
-                            stateList.Add(new RGStateInfo
+                            if (args.Count > 1 && args[1] is { Expression: LiteralExpressionSyntax arg1 })
                             {
-                                StateName = fieldName,
-                                Type = type
-                            });
+                                if (bool.TryParse(arg1.Token.ValueText, out isPlayer))
+                                {
+                                    // not the type, but set our player flag :)
+                                }
+                                else
+                                {
+                                    // is string
+                                    entityTypeName = arg1.Token.ValueText;
+                                }
+                            }
                         }
 
-                        if (stateList.Any())
+                        var hasRGStateAttributes = membersWithRGStateAttribute.Count > 0;
+
+                        var stateMetadata = new List<StateBehaviourPropertyInfo>();
+                        if (hasRGStateAttributes)
                         {
-                            rgStateInfoList.Add(new RGStatesInfo
+                            // do logic for just those attributes
+                            foreach (var member in membersWithRGStateAttribute)
                             {
-                                ClassName = rgStateClassName,
-                                States = stateList
-                            });
+                                var nextEntry = GetStateNameFieldNameAndTypeForMember(true, semanticModel,
+                                    behaviourName, member, includeFlags);
+                                stateMetadata.Add(nextEntry);
+                            }
+                        }
+                        else if (rgStateTypeAttribute != null)
+                        {
+                            if (includeFlags != RGStateTypeAttribute.RGStateIncludeFlags.NONE)
+                            {
+                                // do logic for all the fields based on the type attribute settings
+                                var includeMembers = classDeclaration.Members
+                                    .Where(m =>
+                                        // if it has [RGState explicitly], or is NOT obsolete
+                                        // iow.. obsolete members come through if annotated with [RGState]
```

Comment:
Is there a reason this should be the case? I think it's fine, just curious. I guess if that are explicitly tagging it, they want it

Reply:
My logic.  Obsolete doesn't mean not-supported until they actually remove the field.  They may still want that thing in the state to not break bots until QA can update them.